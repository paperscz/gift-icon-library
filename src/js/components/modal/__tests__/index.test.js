import React from 'react';
import { mount } from 'enzyme';

import { Modal } from '../index';

describe('The Modal component', () => {
  let component;

  const gallery = {
    "key": "test",
    "svgURL": "test.svg",
    "name": "test name",
  };

  it('render the gallery component', () => {
    component = mount(
      <Modal
        closeModal={() => { return; }}
        modal={gallery}
      />
    );
    expect(component).toMatchSnapshot();
  });

  it('render the loading instead of the gallery component', () => {
    component = mount(
      <Modal
        closeModal={() => { return; }}
      />
    );
    expect(component).toMatchSnapshot();
  });

});
