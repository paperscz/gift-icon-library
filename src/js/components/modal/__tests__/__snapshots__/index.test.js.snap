// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Modal component render the gallery component 1`] = `
<Modal
  closeModal={[Function]}
  modal={
    Object {
      "key": "test",
      "name": "test name",
      "svgURL": "test.svg",
    }
  }
>
  <div
    className="overlay"
  >
    <CloseButton
      onClick={[Function]}
    >
      <div
        className="btn-close"
        onClick={[Function]}
      >
        <svg
          height="44"
          viewBox="0 0 475.2 475.2"
          width="44"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M405.6 69.6C360.7 24.7 301.1 0 237.6 0s-123.1 24.7-168 69.6S0 174.1 0 237.6s24.7 123.1 69.6 168 104.5 69.6 168 69.6 123.1-24.7 168-69.6 69.6-104.5 69.6-168-24.7-123.1-69.6-168zm-19.1 316.9c-39.8 39.8-92.7 61.7-148.9 61.7s-109.1-21.9-148.9-61.7c-82.1-82.1-82.1-215.7 0-297.8C128.5 48.9 181.4 27 237.6 27s109.1 21.9 148.9 61.7c82.1 82.1 82.1 215.7 0 297.8z"
          />
          <path
            d="M342.3 132.9c-5.3-5.3-13.8-5.3-19.1 0l-85.6 85.6-85.6-85.6c-5.3-5.3-13.8-5.3-19.1 0-5.3 5.3-5.3 13.8 0 19.1l85.6 85.6-85.6 85.6c-5.3 5.3-5.3 13.8 0 19.1 2.6 2.6 6.1 4 9.5 4s6.9-1.3 9.5-4l85.6-85.6 85.6 85.6c2.6 2.6 6.1 4 9.5 4 3.5 0 6.9-1.3 9.5-4 5.3-5.3 5.3-13.8 0-19.1l-85.4-85.6 85.6-85.6c5.3-5.3 5.3-13.8 0-19.1z"
          />
        </svg>
      </div>
    </CloseButton>
    <Icon
      name="test name"
      svg="test.svg"
    >
      <div
        className="icon"
      >
        <h4>
          test name
        </h4>
        <img
          alt="test name"
          src="test.svg"
        />
        <pre>
          test.svg
        </pre>
      </div>
    </Icon>
  </div>
</Modal>
`;

exports[`The Modal component render the loading instead of the gallery component 1`] = `
<Modal
  closeModal={[Function]}
>
  <span />
</Modal>
`;
